src/linesearch/line_model.jl

```diff
@@ -12,16 +12,16 @@ represents the function ϕ : R → R defined by
 
     ϕ(t) := f(x + td).
 """
-mutable struct LineModel <: AbstractNLPModel
-  meta::NLPModelMeta
+mutable struct LineModel{T, S} <: AbstractNLPModel{T, S}
+  meta::NLPModelMeta{T, S}
   counters::Counters
-  nlp::AbstractNLPModel
+  nlp::AbstractNLPModel{T, S}
```

Comment:
x and below can’t be abstract. Are they of type S?

Reply:
Thanks, this has been fixed by Abel below.